### PR TITLE
New Guns (hopefully no conflicts this time)

### PR DESCRIPTION
**Pull request by TaralGit**
new_weapons

🆑 Taral
- add: new laser guns
- add: Z8 carbine bundle
- add: firecost for most battery guns
- tweak: Inspector has been replaced with the Predator in the uplink and bundle
- tweak: replaced old c20r cev eris sprite with a newer one from the same source
- tweak: Inspector had it's ammo capacity reduced from 7 to 6

### CODE REVIEW

**Comment by TaralGit** [comment]
I'm not touching the old laser gun. That thing is cursed.

**Comment by metalgearsloth** [comment]
Changelog sentence should be readable on its own, the "add" is for the icon.

**Comment by TaralGit** [comment]
> Changelog sentence should be readable on its own, the "add" is for the icon.

Should I delete the changelog?

**Comment by SweptWasTaken** [comment]
Are you pulling these sprites from CEV-Eris master?

For the star pistol you've put the source as "obrez_retro.rsi" which looks like this...

image

For the cog.rsi you've put the source as "energy.rsi" which looks like this...

image

Where are you pulling the pistol sprite from? It looks really awful. If you're looking for a good looking energy pistol sprite you should use the [PDW from Cev-Eris](https://github.com/discordia-space/CEV-Eris/blob/master/icons/obj/guns/energy/pdw.dmi).

image

**Comment by SweptWasTaken** [comment]
> 
> 
> > Changelog sentence should be readable on its own, the "add" is for the icon.
> 
> Should I delete the changelog?

Just shrink it down maybe, doesn't have to be so verbose. Say like "Added more energy weapons". Shrinking the Inspector's ammo should be a tweak.

**Comment by TaralGit** [comment]
> Are you pulling these sprites from CEV-Eris master?
> 
> For the star pistol you've put the source as "obrez_retro.rsi" which looks like this...
> 
> image
> 
> For the cog.rsi you've put the source as "energy.rsi" which looks like this...
> 
> image
> 
> Where are you pulling the pistol sprite from? It looks really awful. If you're looking for a good looking energy pistol sprite you should use the [PDW from Cev-Eris](https://github.com/discordia-space/CEV-Eris/blob/master/icons/obj/guns/energy/pdw.dmi).
> 
> image

Think I'm just going to get rid of the laser pistol, since it's kinda pointless. It's a modified Obrez but I got the wrong link for the meta.json.
Yeah. At first I thought that I should get the PDW in but the problem is that this gun is supposed to also have a Stun setting and I don't know how to code that in.

**Comment by SweptWasTaken** [comment]
> 
> 
> > Are you pulling these sprites from CEV-Eris master?
> > For the star pistol you've put the source as "obrez_retro.rsi" which looks like this...
> > image
> > For the cog.rsi you've put the source as "energy.rsi" which looks like this...
> > image
> > Where are you pulling the pistol sprite from? It looks really awful. If you're looking for a good looking energy pistol sprite you should use the [PDW from Cev-Eris](https://github.com/discordia-space/CEV-Eris/blob/master/icons/obj/guns/energy/pdw.dmi?rgh-link-date=2021-11-06T20%3A55%3A29Z).
> > image
> 
> Think I'm just going to get rid of that laser pistol and replace it with another one I just saw. It's a modified Obrez, as for the COG I got the wrong link for the meta.json but that was fixed.
> 
> At first I thought that I should get the PDW in but the problem is that this gun is supposed to also have a Stun setting and I don't know how to code that in.

I mean it should be very easy looking at the existing weapons. If you can't code it in then don't add it?

**Comment by PaulRitter** [comment]
> I mean it should be very easy looking at the existing weapons. If you can't code it in then don't add it?

I'd like you to stop being mean to people and start being constructive.

For Taral: If you are unsure how to code something, feel free to reach out to us on [discord](https://discord.gg/ss14), namely the #howdoicode channel.

**Comment by TaralGit** [comment]
Well, I'm done with this PR.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml [hunk 1]
```diff
@@ -42,6 +43,141 @@
       steps: 5
       zeroVisible: true
 
+- type: entity
+  name: Makeshift Laser Gun
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml:47]: lowercase the name

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml [hunk 2]
```diff
@@ -42,6 +43,141 @@
...
+      path: /Audio/Weapons/Guns/MagOut/revolver_magout.ogg
+  - type: Appearance
+    visuals:
+    - type: MagVisualizer
+      magState: mag
+      steps: 5
+      zeroVisible: false
+
+- type: entity
+  name: Svalinn Laser Pistol
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml:92]: lowercase name

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml [hunk 3]
```diff
@@ -42,6 +43,141 @@
...
+      path: /Audio/Weapons/Guns/MagOut/revolver_magout.ogg
+  - type: Appearance
+    visuals:
+    - type: MagVisualizer
+      magState: mag
+      steps: 5
+      zeroVisible: false
+
+- type: entity
+  name: COG Laser Carbine
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml:137]: lowercase--ok you get the idea

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Pull request merged**